django/db/models/expressions.py

```diff
@@ -1290,6 +1290,56 @@ def get_db_converters(self, connection):
         ) + self.target.get_db_converters(connection)
 
 
+class Cols(Expression):
+    def __init__(self, alias, targets, sources, output_field):
+        super().__init__(output_field=output_field)
+        self.alias, self.targets, self.sources = alias, targets, sources
+
+    def __len__(self):
+        return len(self.targets)
+
+    def __iter__(self):
+        return iter(self.get_cols())
+
+    def get_cols(self):
+        return [
+            Col(self.alias, target, source)
+            for target, source in zip(self.targets, self.sources)
+        ]
+
+    def get_source_expressions(self):
+        return self.get_cols()
+
+    def set_source_expressions(self, exprs):
+        assert all(isinstance(expr, Col) for expr in exprs)
```

Comment:
We should assert that the `col` aliases still match as well to avoid un-expected relabeling 

```suggestion
        assert all(isinstance(expr, Col) and expr.alias == self.alias for expr in exprs)
```